pkg/build/openbsd.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package build
+
+import (
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"time"
+
+	"github.com/google/syzkaller/pkg/osutil"
+)
+
+type openbsd struct{}
+
+func (ctx openbsd) build(targetArch, vmType, kernelDir, outputDir, compiler, userspaceDir,
+	cmdlineFile, sysctlFile string, config []byte) error {
+	const kernelName = "KCOV"
+	if err := ctx.configure(targetArch, kernelDir, kernelName); err != nil {
+		return err
+	}
+
+	compileDir := fmt.Sprintf("%v/sys/arch/%v/compile/%v", kernelDir, targetArch, kernelName)
+	if err := ctx.make(compileDir, "all"); err != nil {
+		return err
+	}
+
+	for src, dst := range map[string]string{
+		"obj/bsd": "kernel",
+	} {
+		src := filepath.Join(compileDir, src)
+		dst := filepath.Join(outputDir, dst)
+		if err := osutil.CopyFile(src, dst); err != nil {
+			return fmt.Errorf("failed to copy %v -> %v: %v", src, dst, err)
+		}
+	}
+
+	return nil
+}
+
+func (ctx openbsd) clean(kernelDir string) error {
+	return ctx.make(kernelDir, "", "clean")
+}
+
+func (ctx openbsd) configure(targetArch, kernelDir, kernelName string) error {
+	confDir := fmt.Sprintf("%v/sys/arch/%v/conf", kernelDir, targetArch)
+	conf := []byte("include \"arch/amd64/conf/GENERIC\"\npseudo-device kcov 1\n")
```

Comment:
This differs from docs/openbsd/setup.md
And this one actually works, while docs/openbsd/setup.md don't work for me (missing 'make obj` + some other error later). Probably worth updating docs/openbsd/setup.md to keep them in sync.

Reply:
Also the doc contains a type:
```
-$ echo 'include "arch/amd64/conf/KCOV" >>arch/amd64/conf/GENERIC
+$ echo 'include "arch/amd64/conf/KCOV"' >>arch/amd64/conf/GENERIC
```

Reply:
Fixed.